test/e2e/framework/nodes_util.go

```diff
@@ -103,6 +106,46 @@ func masterUpgradeGKE(v string) error {
 	return nil
 }
 
+func masterUpgradeKubernetesAnywhere(v string) error {
+	Logf("Upgrading master to %q", v)
+
+	kaPath := TestContext.KubernetesAnywherePath
+	originalConfigPath := filepath.Join(kaPath, ".config")
+	backupConfigPath := filepath.Join(kaPath, ".config.bak")
+	updatedConfigPath := filepath.Join(kaPath, fmt.Sprintf(".config-%s", v))
+
+	// backup .config to .config.bak
+	if err := os.Rename(originalConfigPath, backupConfigPath); err != nil {
+		return err
+	}
+	defer func() {
+		// revert .config.bak to .config
+		os.Rename(backupConfigPath, originalConfigPath)
+	}()
+
+	// modify config with specified k8s version
+	_, _, err := RunCmd("sed",
+		fmt.Sprintf(`s/kubernetes_version=.*$/kubernetes_version=%s/`, v),
+		backupConfigPath, ">", originalConfigPath)
+	if err != nil {
+		return err
+	}
+
+	// invoke ka upgrade
+	_, _, err = RunCmd("make", "-C", TestContext.KubernetesAnywherePath,
+		"WAIT_FOR_KUBECONFIG=y", "deploy")
```

Comment:
Do we actually run deploy twice? Shouldn't we run `upgrade` or something like that to differ from net-new deployment?

Reply:
This should be the upgrade-master command. Looks like the following on the command line:

make upgrade-master